playground/backend/internal/code_processing/code_processing_test.go

```diff
@@ -1004,3 +1190,354 @@ func syncMapLen(syncMap *sync.Map) int {
+		{
+			name: "Error during set value to RunOutput subKey",
+			mocks: func() {
+				mock.ExpectHSet(pipelineId.String(), marshalCompileOutput, outputMarshal).SetVal(1)
+			},
+			args: args{
+				ctx:          context.Background(),
+				output:       []byte(output),
+				pipelineId:   pipelineId,
+				cacheService: cacheMock,
+			},
+			wantErr: true,
+		},
+		{
+			name: "Error during set value to RunError subKey",
+			mocks: func() {
+				mock.ExpectHSet(pipelineId.String(), marshalCompileOutput, outputMarshal).SetVal(1)
+				mock.ExpectHSet(pipelineId.String(), marshalRunOutput, marshalEmptyString).SetVal(1)
+			},
+			args: args{
+				ctx:          context.Background(),
+				output:       []byte(output),
+				pipelineId:   pipelineId,
+				cacheService: cacheMock,
+			},
+			wantErr: true,
+		},
+		{
+			name: "Error during set value to Logs subKey",
+			mocks: func() {
+				mock.ExpectHSet(pipelineId.String(), marshalCompileOutput, outputMarshal).SetVal(1)
+				mock.ExpectHSet(pipelineId.String(), marshalRunOutput, marshalEmptyString).SetVal(1)
+				mock.ExpectHSet(pipelineId.String(), marshalRunError, marshalEmptyString).SetVal(1)
+			},
+			args: args{
+				ctx:          context.Background(),
+				output:       []byte(output),
+				pipelineId:   pipelineId,
+				cacheService: cacheMock,
+			},
+			wantErr: true,
+		},
+		{
+			name: "Error during set value to Graph subKey",
+			mocks: func() {
+				mock.ExpectHSet(pipelineId.String(), marshalCompileOutput, outputMarshal).SetVal(1)
+				mock.ExpectHSet(pipelineId.String(), marshalRunOutput, marshalEmptyString).SetVal(1)
+				mock.ExpectHSet(pipelineId.String(), marshalRunError, marshalEmptyString).SetVal(1)
+				mock.ExpectHSet(pipelineId.String(), marshalLogs, marshalEmptyString).SetVal(1)
+			},
+			args: args{
+				ctx:          context.Background(),
+				output:       []byte(output),
+				pipelineId:   pipelineId,
+				cacheService: cacheMock,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tt.mocks()
+			if err := processCompileSuccess(tt.args.ctx, tt.args.output, tt.args.pipelineId, tt.args.cacheService); (err != nil) != tt.wantErr {
+				t.Errorf("processCompileSuccess() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func Test_readGraphFile(t *testing.T) {
+	pipelineLifeCycleCtx, _ := context.WithTimeout(context.Background(), 1*time.Second)
+	type args struct {
+		pipelineLifeCycleCtx context.Context
+		backgroundCtx        context.Context
+		cacheService         cache.Cache
+		graphFilePath        string
+		pipelineId           uuid.UUID
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "Successfully saving the prepared graph to the cache",
+			args: args{
+				pipelineLifeCycleCtx: pipelineLifeCycleCtx,
+				backgroundCtx:        context.Background(),
+				cacheService:         cacheService,
+				graphFilePath:        graphFilePath,
+				pipelineId:           uuid.New(),
+			},
```

Comment:
What happens when we don't have the file with a graph? Does any error raise?

Reply:
There are no errors raised. If we don't have a graph file, we'll just skip saving it to the cache.